links/gql_dio_link/lib/src/dio_link.dart

```diff
@@ -91,7 +94,10 @@ class DioLink extends Link {
       },
     );
 
-    if (dioResponse.statusCode >= 300 ||
```

Comment:
It might be that this check is always false (and not needed) if I understood correctly that Dio throws error (instead of returning a response) if status code is >= 300. 

Anyway, adding that optional check of valid status codes here would not change behaviour at all if you don't provide the optional status codes in constructor.